component-library/components/generic/form/phone-input/phone-input.eleventy.liquid

```diff
@@ -0,0 +1,24 @@
+{% assign_local c = "c-phone-input" %}
+<div class="{{ c }}">
+    <label class="{{ c }}__label" for="{{ id }}">{{ label }}{% if required %} *{% endif %}</label>
+    <input class="{{ c }}__input"
+        type="tel" id="{{ id }}" 
```

Comment:
type=tel is only used to bring up the numpad on supported devices and browsers. It does not perform any validation / restriction on user input, so someone can still enter text. You can get around this by using a pattern to make the browser validate it.

see here for the pattern: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/tel
You can also write a regex for it.